DEFEDIT-1206 Optionally track active item in asset browser

ragnard-king opened the pull request:
### User facing changes

Fixes https://github.com/defold/editor2-issues/issues/1357

- There's now a new setting "Track Active Item in Asset Browser". When
  enabled, the resource being edited will also be selected and focused
  in the asset browser.

### Technical changes

- Separate production of tree root and update of `TreeView` and make
  update smarter depending on what has actually changed.
- New preference "asset-browser-track-active-item?"
- Always scroll to selection in asset browser

ragnard-king committed:
DEFEDIT-1206 Optionally track active item in asset browser

* User facing changes

- There's now a new setting "Track Active Item in Asset Browser". When
  enabled, the resource being edited will also be selected and focused
  in the asset browser.

* Technical changes

- Separate production of tree root and update of TreeView and make
  update smarter depending on what has actually changed.
- New preference "asset-browser-track-active-item?"
- Always scroll to selection in asset browser

matgis-king reviewed (changes requested):
Reviewed! Code looks good, but it doesn't seem to work if I select the tab of a resource that is located more than one directory below the project root. I.e. it works on `/utils/url.lua`, but not on `/game/gui/coin.lua` unless I expand the `/game` directory first.

ragnard-king committed:
DEFEDIT-1206 Review changes

- rename new preference to "Track Active Tab in Asset Browser"
- fix bug in auto-expand that failed to expand a hierarchy deeper than
  1 level properly.
